timemachine/fe/single_topology.py

```diff
@@ -1133,6 +1255,117 @@ def __init__(self, mol_a: Chem.Mol, mol_b: Chem.Mol, core: NDArray, forcefield:
         assert_default_system_constraints(self.src_system)
         assert_default_system_constraints(self.dst_system)
 
+        # align tuples and compute interpolation schedules for each term
+        self.aligned_bond = self._align_bonds()
+        self.aligned_angle = self._align_angles()
+        self.aligned_proper = self._align_propers()
+        self.aligned_improper = self._align_impropers()
+        self.aligned_chiral_atom = self._align_chiral_atoms()
+        self.aligned_nonbonded_pairlist = self._align_nonbonded_pairlist()
+
+    def _align_bonded_term(self, align_fn, assign_min_max_fn, src_potential, dst_potential):
+        aligned_tuples = align_fn(
+            src_potential.potential.idxs,
+            src_potential.params,
+            dst_potential.potential.idxs,
+            dst_potential.params,
+        )
+        aligned_idxs = np.array([x[0] for x in aligned_tuples], dtype=np.int32)
+        aligned_src_params = np.array([x[1] for x in aligned_tuples], dtype=np.float64)
+        aligned_dst_params = np.array([x[2] for x in aligned_tuples], dtype=np.float64)
+        aligned_mins, aligned_maxes = assign_min_max_fn(aligned_tuples)
+
+        return aligned_idxs, aligned_src_params, aligned_dst_params, aligned_mins, aligned_maxes
+
+    def _align_bonds(self):
+        idxs, src_params, dst_params, mins, maxes = self._align_bonded_term(
+            interpolate.align_harmonic_bond_idxs_and_params,
+            self._assign_bond_idxs_min_max,
+            self.src_system.bond,
+            self.dst_system.bond,
+        )
+        idxs = idxs.reshape(-1, 2)
+        src_params = src_params.reshape(-1, 2)
+        dst_params = dst_params.reshape(-1, 2)
```

Comment:
Are these reshapes necessary? (Looks to me like the arrays returned by `_align_bonded_term` already have the correct shapes, but probably missing something.

Reply:
Yes, this was needed to fix a couple of tests when propers/chiral volumes were missing.  `_align_bonded_term ` calls:

```
  aligned_src_params = jnp.array([x[1] for x in aligned_tuples], dtype=np.float64)
  aligned_dst_params = jnp.array([x[2] for x in aligned_tuples], dtype=np.float64)
```

if `len(aligned_tuples) == 0` then this can return a `jnp.array` with the wrong dimensions.

Reply:
Ah, got it. Empty arrays strike again.